keepalived service: init

mbrgm opened the pull request.

joachifm commented on nixos/modules/services/networking/keepalived.nix:
```diff
@@ -0,0 +1,53 @@
+    systemd.services.keepalived = {
+      description = "Keepalive Daemon (LVS and VRRP)";
+      after = [ "network.target" "syslog.target" ];
+      wants = [ "network.target" ];
```
Explicit dependency on `network.target` is probably redundant, it'll be part of normal bootup anyway. Also, `network.target` is very weakly defined. All you can be sure of is that your service will be stopped *before* network on shutdown ... If you intended to require network connectivity or something, you'd depend on and order after `network-online.target`, that at least is semantically more correct.

joachifm commented on nixos/modules/services/networking/keepalived.nix:
```diff
@@ -0,0 +1,53 @@
+
+      extraConfig = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
```
I think this default value will fail to evaluate (unless explicitly converted to a string with `toString`). I think just using an empty string here is okay anyway.

mbrgm committed:
keepalived service: init

mbrgm commented:
Note that build breakage did not originate from this change.

mbrgm commented:
@fpletz From my side, you can remove the WIP label -- the last commit is the final implementation.

joachifm commented on nixos/modules/services/networking/keepalived/default.nix:
```diff
@@ -0,0 +1,253 @@
+        };
+
+        socket = mkOption {
+          type = types.nullOr types.string;
```
I suspect you want to use `types.str`

mbrgm replied:
Yup, you're right!

joachifm commented on nixos/modules/services/networking/keepalived/vrrp-options.nix:
```diff
@@ -0,0 +1,141 @@
+      '';
+    };
+
+    authPass = mkOption {
```
Looks like this ends up in the Store (and so will be world-readable), is that a problem?

mbrgm replied:
I looked this up once more and found that authentication was deprecated in [RFC3768](https://www.ietf.org/rfc/rfc3768.txt) and is only there for backwards compatibility. I'm going to remove the whole authentication section so people don't use a deprecated mechanism.

mbrgm committed:
Remove authentication options

Authentication was deprecated in
[RFC3768](https://www.ietf.org/rfc/rfc3768.txt).

mbrgm committed:
Replace types.string with types.str

mbrgm commented:
@joachifm I added two commits addressing the issues you mentioned. Does this match what you had in mind?

joachifm commented:
Thank you
